File: go/cryptowallet/wallet.go
package cryptowallet

type Wallet struct { Coin, Address string; WatchOnly bool }
var SupportedCoins = []string{"BTC","BCH","LTC","DOGE","ETH","ETC","SOL","ADA","XRP","DOT","AVAX"}
type Transaction struct { State, Coin, From, To, Amount string }
func (w Wallet) Transaction(to, amount string) Transaction { return Transaction{"awaiting_confirmation",w.Coin,w.Address,to,amount} }

File: mobile/kotlin/src/main/kotlin/bizx/crypto/Wallet.kt
package bizx.crypto

data class Wallet(val coin: String, val address: String, val watchOnly: Boolean = true) {
    companion object { val supportedCoins = setOf("BTC","BCH","LTC","DOGE","ETH","ETC","SOL","ADA","XRP","DOT","AVAX") }
    init { require(coin in supportedCoins) }
    fun transaction(to: String, amount: String) = mapOf("state" to "awaiting_confirmation", "coin" to coin, "from" to address, "to" to to, "amount" to amount)
}
